apps/src/lib/kits/maker/boards/FakeMBBoard.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Fake MicroBit Board for running Maker Toolkit apps without a MicroBit board
+ * attached.
+ * Attaches fake, no-op components to the interpreter.
+ * @extends EventEmitter
+ * @implements MakerBoard
+ */
+export default class FakeMBBoard extends EventEmitter {
+  constructor() {
+    super();
+    this.boardClient_ = new MBFirmataClientStub();
+  }
+
+  /**
+   * Open a connection to the board on its configured port.
+   * @returns {Promise} resolved when the board is ready to use.
+   */
+  connect() {
+    return Promise.resolve();
+  }
+
+  /**
+   * Disconnect and clean up the board controller and all components.
+   * @returns {Promise}
+   */
+  destroy() {
+    return Promise.resolve();
+  }
+
+  openSerialPort() {}
+
+  /**
+   * Marshals the board component controllers and appropriate constants into the
+   * given JS Interpreter instance so they can be used by student code.
+   *
+   * @param {JSInterpreter} jsInterpreter
+   */
+  installOnInterpreter(jsInterpreter) {
+    const constructors = {
+      MicroBitButton: FakeMicroBitButton,
+      LedScreen: FakeLedScreen,
+      Accelerometer: FakeAccelerometer,
+      MicroBitThermometer: FakeMicroBitThermometer,
+      Compass: FakeCompass,
+      LightSensor: FakeLightSensor,
+      ExternalButton: FakeExternalButton,
+      ExternalLed: FakeExternalLed,
+      CapacitiveTouchSensor: FakeCapacitiveTouchSensor
+    };
+
+    for (const constructorName in constructors) {
+      const constructor = constructors[constructorName];
+      jsInterpreter.addCustomMarshalObject({instance: constructor});
+      jsInterpreter.createGlobalProperty(constructorName, constructor);
+    }
+
+    const components = {
+      buttonA: new FakeMicroBitButton(),
+      buttonB: new FakeMicroBitButton(),
+      ledScreen: new FakeLedScreen(),
+      tempSensor: new FakeMicroBitThermometer(),
+      accelerometer: new FakeAccelerometer(),
+      lightSensor: new FakeLightSensor(),
+      compass: new FakeCompass(),
+      board: new FakeMBFirmataWrapper()
+    };
+
+    for (const componentName in components) {
+      const component = components[componentName];
+      jsInterpreter.createGlobalProperty(componentName, component);
+    }
+  }
+
+  /**
+   * @param {number} pin
+   * @param {number} modeConstant
+   */
+  pinMode(pin, modeConstant) {}
+
+  /**
+   * @param {number} pin
+   * @param {number} value
+   */
+  digitalWrite(pin, value) {}
+
+  /**
+   * @param {number} pin
+   * @param {function.<number>} callback
+   */
+  digitalRead(pin, callback) {
+    setTimeout(() => callback(0), 0);
+  }
+
+  /**
+   * @param {number} pin
+   * @param {number} value
+   */
+  analogWrite(pin, value) {}
+
+  /**
+   * @param {number} pin
+   * @param {function.<number>} callback
+   */
+  analogRead(pin, callback) {
+    setTimeout(() => callback(0), 0);
+  }
+
+  /**
+   * @return {boolean} whether a real board is connected
+   */
+  boardConnected() {
+    return false;
+  }
+
+  createLed(pin) {
```

Comment:
Should these also have params, like the ones above?

Reply:
(sorry, I meant should they have doc comments regarding their params, like the ones above)

Reply:
Sure - and I'll add to `FakeCPBoard.js` as well.